lib/sshutils/server.go

```diff
@@ -125,15 +127,25 @@ func (s *Server) Wait() {
 
 // Close closes listening socket and stops accepting connections
 func (s *Server) Close() error {
-	return s.listener.Close()
+	s.askedToClose = true
+	if s.listener != nil {
+		return s.listener.Close()
+	}
+	return nil
 }
 
 func (s *Server) acceptConnections() {
 	defer s.notifyClosed()
-	log.Infof("%v ready to accept connections", s.Addr())
+	addr := s.Addr()
+	log.Infof("%v ready to accept connections", addr)
 	for {
 		conn, err := s.listener.Accept()
 		if err != nil {
+			if s.askedToClose {
```

Comment:
askedToClose is a cosmetic flag which does not affect the execution logic, it only suppresses unwanted log messages. It is not a race because it's set before closing a socket (or any other blocking call) and checked after the socket (or similar object) is closed.

in cases when there wasn't a synchronized resource (like socket) I used your `utils.CloseBroadcaster`